riscv/execute.cc

```diff
@@ -244,6 +237,16 @@ void processor_t::step(size_t n)
           abort();
       }
     }
+    catch (wait_for_interrupt_t &t)
+    {
+      // Return to the outer simulation loop, which gives other devices/harts a
+      // chance to generate interrupts.
+      //
+      // In the debug ROM this prevents us from wasting time looping, but also
+      // allows us to switch to other threads only once per idle loop in case
+      // there is activity.
+      return;
```

Comment:
This approach looks good, but I think this control path fails to update `state.minstret`.  Instead of doing a `return;` you can do `n = instret;`, just like the case for `catch(trap_t& t)` does.  This will cause the loop to terminate normally.

Reply:
Good point. I've fixed that.